From b857d3affccf870501f7b9de34f837a1a2575046 Mon Sep 17 00:00:00 2001
From: Patrick Steuer <patrick.steuer@de.ibm.com>
Date: Fri, 15 Feb 2019 22:59:09 +0100
Subject: [PATCH 06/25] s390x assembly pack: fix formal interface bug in chacha
 module

Signed-off-by: Patrick Steuer <patrick.steuer@de.ibm.com>

Reviewed-by: Tim Hudson <tjh@openssl.org>
Reviewed-by: Richard Levitte <levitte@openssl.org>
(Merged from https://github.com/openssl/openssl/pull/8257)

(cherry picked from commit b2b580fe445e064da50c13d3e00f71022da16ece)
---
 crypto/chacha/asm/chacha-s390x.pl | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/crypto/chacha/asm/chacha-s390x.pl b/crypto/chacha/asm/chacha-s390x.pl
index 895765e1c4..2843bb1eb6 100755
--- a/crypto/chacha/asm/chacha-s390x.pl
+++ b/crypto/chacha/asm/chacha-s390x.pl
@@ -225,7 +225,7 @@ LABEL	("ChaCha20_ctr32");
 	larl	("%r1","OPENSSL_s390xcap_P");
 
 	lghi	("%r0",64);
-&{$z?	\&cgr:\&cr}	($len,"%r0");
+&{$z?	\&clgr:\&clr}	($len,"%r0");
 	jle	("_s390x_chacha_novx");
 
 	lg	("%r0","S390X_STFLE+16(%r1)");
-- 
2.25.1

